crates/store/src/state_db.rs

```diff
@@ -1,57 +1,87 @@
 //! State DB
 
-use std::{cell::RefCell, collections::HashSet};
+use std::{cell::RefCell, collections::HashSet, mem::size_of_val};
 
 use crate::{smt_store_impl::SMTStore, traits::KVStore, transaction::StoreTransaction};
 use gw_common::{error::Error as StateError, smt::SMT, state::State, H256};
 use gw_db::schema::{
     Col, COLUMN_ACCOUNT_SMT_BRANCH, COLUMN_ACCOUNT_SMT_LEAF, COLUMN_DATA, COLUMN_SCRIPT,
 };
-use gw_db::{error::Error, iter::DBIter, DBVector, IteratorMode};
+use gw_db::{error::Error, iter::DBIter, DBRawIterator, IteratorMode};
 use gw_traits::CodeStore;
 use gw_types::{bytes::Bytes, packed, prelude::*};
 
-pub struct StateDBVersion(String);
+const FLAG_DELETE_VALUE: u8 = 0;
+const BLOCK_NUMBER_ZERO: u64 = 0;
+const TX_INDEX_ZERO: u32 = 0;
+
+pub struct StateDBVersion {
+    block_hash: Option<H256>,
+    tx_index: Option<u32>,
+}
 
 impl StateDBVersion {
-    /// TODO implement this
-    pub fn from_block_hash(_block_hash: H256) -> Self {
-        Self(Default::default())
+    pub fn from_genesis() -> Self {
+        StateDBVersion {
+            block_hash: None,
+            tx_index: None,
+        }
     }
 
-    /// TODO implement this
-    pub fn from_genesis() -> Self {
-        Self(Default::default())
+    pub fn from_block_hash(block_hash: H256) -> Self {
+        StateDBVersion {
+            block_hash: Some(block_hash),
+            tx_index: None,
+        }
+    }
+
+    pub fn from_tx_index(block_hash: H256, tx_index: u32) -> Self {
+        StateDBVersion {
+            block_hash: Some(block_hash),
+            tx_index: Some(tx_index),
+        }
+    }
+
+    pub fn is_genesis_version(&self) -> bool {
+        self.block_hash.is_none() && self.tx_index.is_none()
     }
 }
 
-// TODO implement this
 pub struct StateDBTransaction {
     inner: StoreTransaction,
-    _version: StateDBVersion,
+    block_num: u64,
```

Comment:
```suggestion
    block_number: u64,
```

keep a consistent naming style